types/ember/index.d.ts

```diff
@@ -315,6 +314,121 @@ interface RenderOptions {
     view?: string;
 }
 
+interface ViewMixin {
+    // methods
+    $: JQueryStatic;
+    rerender(): void;
+    // properties
+    attributeBindings: string[];
+    element: Element;
+    elementId: string;
+    tagName: string;
+    // events
+    didInsertElement(): void;
+    willClearRender(): void;
+    willDestroyElement(): void;
+    willInsertElement(): void;
+}
+const ViewMixin: Ember.Mixin<ViewMixin>;
+
+/**
+Ember.CoreView is an abstract class that exists to give view-like behavior to both Ember's main
+view class Ember.Component and other classes that don't need the full functionality of Ember.Component.
+
+Unless you have specific needs for CoreView, you will use Ember.Component in your applications.
+**/
+class CoreView extends Ember.Object.extend(Ember.Evented, ActionHandler) {}
+interface ActionSupport {
+    sendAction(action: string, ...params: any[]): void;
+}
+const ActionSupport: ActionSupport;
+
+interface ClassNamesSupport {
```

Comment:
Where does `ClassNamesSupport` and `ActionSupport` come from?

Reply:
https://emberjs.com/api/ember/2.14/classes/Ember.ClassNamesSupport
https://emberjs.com/api/ember/2.14/classes/Ember.ActionSupport

Both are in the 'uses' list of https://emberjs.com/api/ember/2.14/classes/Ember.Component

Reply:
Oh, ok, I was testing on ember 2.9, those must have been introduced later.